main.go

```diff
@@ -1,34 +1,76 @@
 	"time"
 
-	// "os"
-	// "os/signal"
 	"strings"
 	"sync"
 
-	// "time"
-	"github.com/pkg/errors"
-
 	"github.com/Shopify/sarama"
-	"github.com/quipo/statsd"
+	"github.com/alecthomas/kingpin"
+	"github.com/pkg/errors"
 	"github.com/segmentio/go-log"
-	// "github.com/wvanbergen/kazoo-go"
+	"gopkg.in/alexcesaro/statsd.v2"
 )
 
 var (
-	brokers      = flag.String("brokers", "127.0.0.1:12008", "Kafka addresses (e.g. host1:9092,host2:9092)")
-	statsdAddr   = flag.String("statsd-addr", "127.0.0.1:8125", "Statsd address")
-	statsdPrefix = flag.String("statsd-prefix", "kafka.", "Statsd prefix")
-	interval     = flag.Int("refresh-interval", 5, "Interval to refresh offset lag in seconds")
-	useTags      = flag.Bool("use-tags", true, "Use tags if your StatsD client supports them (like DataDog and InfluxDB)")
+	brokers      = kingpin.Flag("broker", "A kafka broker to connect to.  Specify multiple times for multiple brokers. (e.g. host1:9092)").HintOptions("host1:9092").Short('b').Envar("KSTATSD_BROKERS").Required().Strings()
+	statsdAddr   = kingpin.Flag("statsd-addr", "Statsd address").Short('s').Default("127.0.0.1").Envar("KSTATSD_STATSD_ADDR").String()
+	statsdPort   = kingpin.Flag("statsd-port", "Statsd port").Short('P').Default("8125").Envar("KSTATSD_STATSD_PORT").String()
+	statsdPrefix = kingpin.Flag("statsd-prefix", "Statsd prefix").Short('p').Envar("KSTATSD_STATSD_PREFIX").String()
+	interval     = kingpin.Flag("refresh-interval", "Interval to refresh offset lag in seconds").Short('i').Default("5").Envar("KSTATSD_INTERVAL").Int()
+	tagType      = kingpin.Flag("tag-format", "Format to use when encoding tags (Options: none, influxdb, datadog)").HintOptions(statsdTagOptionsEnum()...).Default("none").Envar("KSTATSD_USE_TAGS").Enum(statsdTagOptionsEnum()...)
+	includeTags  = kingpin.Flag("tag", "Tags to include.  Specify multiple times for multiple tags. (e.g. tagname:value)").HintOptions("tagname:value").Envar("KSTATSD_TAGS").Strings()
 )
 
+var statsdTagFormat = map[string]statsd.TagFormat{
+	"influxdb": statsd.InfluxDB,
+	"datadog":  statsd.Datadog,
+	"none":     0,
+}
+
+func statsdTagOptionsEnum() []string {
+	res := make([]string, 0, len(statsdTagFormat))
+	for k := range statsdTagFormat {
+		res = append(res, k)
+	}
+	return res
+}
+
+func newStatsdClient() (*statsd.Client, error) {
+	tags := make([]string, 0, len(*includeTags)*2)
+	for _, tag := range *includeTags {
+		splitTag := strings.SplitN(tag, ":", 2)
+		tags = append(tags, splitTag...)
+	}
+
+	opts := []statsd.Option{
+		statsd.Address(strings.Join([]string{*statsdAddr, *statsdPort}, ":")),
+		statsd.ErrorHandler(func(err error) {
+			log.Error("Statsd error: %s", err)
+		}),
+	}
+
+	if *statsdPrefix != "" {
+		opts = append(opts, statsd.Prefix(*statsdPrefix))
+	}
+
+	if len(tags) > 0 {
+		opts = append(opts, statsd.Tags(tags...))
+	}
+
+	tagFormat := statsdTagFormat[*tagType]
+	if tagFormat != 0 {
+		opts = append(opts, statsd.TagsFormat(tagFormat))
+	}
+
+	return statsd.New(opts...)
```

Comment:
Should this be `statsd.New(), nil` or does statsd.New return an `(instance, error)` itself?

Reply:
`statsd.New` returns `(instance, error)`